resources/resource_transformers/postcss/postcss.go

```diff
@@ -140,69 +143,88 @@ func (t *postcssTransformation) Key() internal.ResourceTransformationKey {
 // npm install -g autoprefixer
 func (t *postcssTransformation) Transform(ctx *resources.ResourceTransformationCtx) error {
 
-	const localPostCSSPath = "node_modules/.bin/"
+	localCommand := ""
+	localConfig := ""
 	const binaryName = "postcss"
 
-	// Try first in the project's node_modules.
-	csiBinPath := filepath.Join(t.rs.WorkingDir, localPostCSSPath, binaryName)
-
-	binary := csiBinPath
-
-	if _, err := exec.LookPath(binary); err != nil {
-		// Try PATH
-		binary = binaryName
-		if _, err := exec.LookPath(binary); err != nil {
-			// This may be on a CI server etc. Will fall back to pre-built assets.
-			return herrors.ErrFeatureNotAvailable
-		}
-	}
-
 	var configFile string
-	logger := t.rs.Logger
-
 	if t.options.Config != "" {
 		configFile = t.options.Config
 	} else {
 		configFile = "postcss.config.js"
 	}
 
-	configFile = filepath.Clean(configFile)
-
-	// We need an abolute filename to the config file.
-	if !filepath.IsAbs(configFile) {
-		// We resolve this against the virtual Work filesystem, to allow
-		// this config file to live in one of the themes if needed.
-		fi, err := t.rs.BaseFs.Work.Stat(configFile)
-		if err != nil {
-			if t.options.Config != "" {
-				// Only fail if the user specificed config file is not found.
-				return errors.Wrapf(err, "postcss config %q not found:", configFile)
+	nodePath := ""
+	binPath := ""
+	for idx, mod := range t.rs.PathSpec.Paths.AllModules {
+		dir := mod.Dir()
+		if localCommand == "" {
+			commandPath := path.Join(dir, "node_modules", ".bin", binaryName)
```

Comment:
So, the idea is good, but this assumes that `npm install` has been run -- which Hugo cannot do as a general thing (modules pulled in via Hugo Modules are read-only)

Reply:
Interesting. That would assume that npm folders need to either be published as part of the module, or updated in some other way. It does make sense, but that will make a lot of things more difficult if modules' js files require things which are not necessarily available on the project node_modules and package.json. That's in interesting dilemma :)